torch/csrc/jit/script/compiler.cpp

```diff
@@ -2767,17 +2767,15 @@ struct to_ir {
         if (type_hint && type_hint->kind() == TypeKind::ListType) {
           elem_type = type_hint->expect<ListType>()->getElementType();
         } else if (!values.empty()) {
-          elem_type = values.at(0)->type();
-        }
-
-        // Tensors are special because they have dymnamic properties. So any
-        // list containing tensors should be typed with the unified typeof all
-        // the elements.
-        if (elem_type->isSubtypeOf(TensorType::get())) {
-          for (const auto& value : values) {
-            elem_type = unifyTypes(elem_type, value->type()).value();
+          std::stringstream ss;
+          auto types = fmap(values, [](const Value* v) { return v->type(); });
+          auto maybe_elem_type = unifyTypeList(std::move(types), ss);
```

Comment:
You don't have to std::move here because unifyTypeList takes in an `at::ArrayRef` (and it might be wrong to move into a function that takes in a ref as well).